axlearn/common/trainer.py

```diff
@@ -853,7 +854,7 @@ def _prepare_training(self, prng_key: Tensor) -> bool:
 
         # Attempt to restore the latest checkpoint, which may contain a saved `_input_iter`.
         self.restore_checkpoint(restore_step=None)
-
+        self._maybe_record_event(measurement.Event.START_TRAINING_PREPARATION)
```

Comment:
Should this be moved to before `self.restore_checkpoint`?

Reply:
Its likely that restore_checkpoint might need to be wrapped around the checkpoint logging.

Since you have maximum context of underlying operations, if you could review & provide guidance on the location of each of the Recorder API calls for accelerator initialization, training preparation and data loading, that would be great.

Reply:
checkpoint restore is part of training preparation. So I think it makes sense to include it.
We should just properly map "TRAINING_PREPARATION" to  "_prepare_training" function. 

This is also aligned with the design doc you shared earlier

> This is the time spent on the creation of checkpoint managers, checkpoint loading, running
mesh and model optimizers etc. 


Separately,  it would be nice to add "checkpoint" specific events in the near future.

But does Badput support nested structure? What happens if we record checkpoint_restore related event, which is part of "training_preparation", will calculation be messed up?